hnn_core/tests/test_network.py

```diff
@@ -128,20 +127,121 @@ def test_network_cell_positions():
                     net.pos_dict['origin'][idx])
 
 
-def test_network():
-    """Test network object."""
+def test_network_drives():
+    """Test manipulation of drives in the network object."""
     with pytest.raises(TypeError, match='params must be an instance of dict'):
         Network('hello')
     params = read_params(params_fname)
-    # add rhythmic inputs (i.e., a type of common input)
-    params.update({'input_dist_A_weight_L2Pyr_ampa': 1.4e-5,
-                   'input_dist_A_weight_L5Pyr_ampa': 2.4e-5,
-                   't0_input_dist': 50,
-                   'input_prox_A_weight_L2Pyr_ampa': 3.4e-5,
-                   'input_prox_A_weight_L5Pyr_ampa': 4.4e-5,
-                   't0_input_prox': 50})
-
-    net = jones_2009_model(deepcopy(params), add_drives_from_params=True)
+    net = jones_2009_model(params, legacy_mode=False)
+
+    # add all drives explicitly and ensure that the expected number of drive
+    # cells get instantiated for each case
+    n_drive_cells_list = list()
+
+    # use erp drive sequence
+    n_drive_cells = 'n_cells'
+    n_drive_cells_list.append(n_drive_cells)
+    drive_weights = dict()
+    drive_weights['evdist1'] = {'L2_basket': 0.01, 'L2_pyramidal': 0.02,
+                                'L5_pyramidal': 0.03}
+    drive_delays = dict()
+    drive_delays['evdist1'] = {'L2_basket': 0.1, 'L2_pyramidal': 0.2,
+                               'L5_pyramidal': 0.3}
```

Comment:
These values are specified explicitly at the get-go and then referenced later to ensure their integrity is maintained throughout the `Network` object.